node/resolvers/auth/index.ts

```diff
@@ -14,15 +16,24 @@ const makeRequest = async (ctx, url) => {
 }
 
 export const mutations = {
-  sendEmailVerification: async (_, args, { vtex: ioContext }) => {
-    const { data: { authenticationToken } } = await makeRequest(ioContext, paths.getTemporaryToken())
-    await makeRequest(ioContext, paths.sendEmailVerification(args.email, authenticationToken))
-    return { authToken: authenticationToken }
+  sendEmailVerification: async (_, args, { vtex: ioContext, response }) => {
+    const { data, status } = await makeRequest(ioContext, paths.getTemporaryToken(ioContext.account, ioContext.account))
+    if (!data.authenticationToken) {
+      throw new ResolverError(`ERROR ${data}`, status)
+    }
+    await makeRequest(ioContext, paths.sendEmailVerification(args.email, data.authenticationToken))
+    response.set('Set-Cookie', serialize('temporarySession', data.authenticationToken, { path: '/', }))
+    return data.authenticationToken ? true : false
   },
 
-  signIn: async (_, args, { vtex: ioContext, request: { headers: { cookie } }, response }) => {
-    const { fields: { email, authToken, code } } = args
-    const { data: { authCookie } } = await makeRequest(ioContext, paths.signIn(email, authToken, code))
-    response.set('Set-Cookie', serialize(authCookie.Name, authCookie.Value, { httpOnly: true }))
+  accessKeySignIn: async (_, args, { vtex: ioContext, request: { headers: { cookie } }, response }) => {
+    // FIXME: Get temporarySession in cookies instead of args. @brunojdo - 08/06/2018 
+    const { fields: { email, temporarySession, code } } = args
+    const { data, status } = await makeRequest(ioContext, paths.accessKeySignIn(email, temporarySession, code))
+    if (!data.authCookie) {
+      throw new ResolverError(`ERROR ${data.authStatus}`, status)
+    }
+    response.set('Set-Cookie', serialize(data.authCookie.Name, data.authCookie.Value, { httpOnly: true }))
```

Comment:
Add the Secure flag to the cookie by passing the following config object:
```js
{
  httpOnly: true,
  secure: true
}
```
```js
response.set('Set-Cookie', serialize(data.authCookie.Name, data.authCookie.Value, { httpOnly: true,  secure: true }))
```

The flag Secure says this cookie can only be set or sent if inside a HTTPS domain.